File: common/scripts/.bin/checkout.lua
#!/usr/bin/luajit

local branches = {}
local repoPath = arg[1] 
for branch in io.popen('git -C ' .. repoPath  .. ' branch'):lines() do
	table.insert(branches, branch )
end 

local chosen = rofiMenu(branches, {prompt = 'git checkout', width = '25ch'})

local status = os.execute('git -C ' .. repoPath .. ' checkout ' .. chosen)
if status == 0 then
	notify('Checked out to ' .. chosen)	
else
	notifyError('Could not checkout out to ' .. chosen)
end

File: common/scripts/.bin/checkout.clj
#!/usr/bin/env bb

;(ns checkout
(require '[babashka.process :as ps :refer [$ shell process sh]]
         '[clojure.string :as str]
         '[clojure.core.match :refer [match]]
         )
;)

(comment
  (require '[portal.api :as p])
  (add-tap #'p/submit)
  (def p (p/open {:launcher :intellij}))
  (tap> :hello)
  (load-file (str (System/getenv "HOME") "/Documents/dotfiles/common/scripts/.bin/clj/init.clj"))
  )

(defn- run-rofi [dir]
  (when-let [selected (as-> dir p
                 (sh "git" "-C" p "branch")
                 (:out p)
                 (rofi-menu! p {:prompt "git checkout", :width "25ch"})
                 (:out p)
                 (first p))]
    (str/trim selected)))

(let [dir (first *command-line-args*)]
  (if (and dir (fs/directory? dir))
    (if (->> (run-rofi dir)
              (sh "git" "-C" dir "checkout")
              (:exit)
              (zero?))
      (notify! "Switched branch")
      (notify-error! "Couldn't switch branch"))
    (notify-error! "Invalid directory path")))

(comment
  )
